src/lib/Libifl/pbsD_connect.c

```diff
@@ -638,7 +652,74 @@ __pbs_connect_extend(char *server, char *extend_data)
 		return -1;
 	}
 
-	return connect_to_servers(server_name, server_port, extend_data);
+	nsvrs = get_num_servers();
+
+	if (nsvrs == 1 && pbs_conf.pbs_primary && pbs_conf.pbs_secondary) {	
+		/* failover configuered ...   */	
+		if (is_same_host(server, pbs_conf.pbs_primary)) {	
+			have_alt = 1;	
+			/* We want to try the one last seen as "up" first to not   */	
+			/* have connection delays.   If the primary was up, there  */	
+			/* is no .pbsrc.NAME file.  If the last command connected  */	
+			/* to the Secondary, then it created the .pbsrc.USER file. */	
+
+			/* see if already seen Primary down */	
+#ifndef CHECK_FILE	
```

Comment:
I think we could avoid ifndef block. Keep it as the default. Override the bheavior if CHECK_FILE is defined.